### PR TITLE
Add Janus removeStreams endpoint

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@fvovan please review 

@alexispeter @njam wdyt about removing `final` from `CM_Model_Abstract::delete()` method? :panda_face:

File: library/CM/Janus/RpcEndpoints.php [hunk 1]
```diff
@@ -178,6 +178,27 @@ public static function rpc_removeStream($serverKey, $channelKey, $streamKey) {
     }
 
     /**
+     * @param string $serverKey
+     * @return bool
+     * @throws CM_Exception_AuthFailed
+     */
+    public static function rpc_removeAllStreams($serverKey) {
```
Comment (fvovan) [inline library/CM/Janus/RpcEndpoints.php]: Why it's called `rpc_removeAllStreams ` when we actually remove stream channels?

File: library/CM/Model/Abstract.php [hunk 2]
```diff
@@ -101,7 +101,7 @@ public function commit($useReplace = null) {
         $this->_autoCommit = true;
     }
 
-    final public function delete() {
```
Comment (fvovan) [inline library/CM/Model/Abstract.php]: was it done by accident or to allow mocking?

**Comment by njam** [comment]
Agree to removing `final`!

File: library/CM/Janus/RpcEndpoints.php (hunk 1 above)
Comment (tomaszdurka) [inline library/CM/Janus/RpcEndpoints.php]: StreamChannel is concept which is not really known to cm-janus. Cm-janus only publishes, subscribes (creates stream) or removes stream.

File: library/CM/Model/Abstract.php (hunk 2 above)
Comment (tomaszdurka) [inline library/CM/Model/Abstract.php]: Yeah. This is due to Mocka (nor any other mocking framework) not able to extend final methods.

I have seen ways using php runkit - but it's quite dirty as it modifies original behaviour.

**Comment by tomaszdurka** [comment]
@fvovan please confirm.

**Comment by fvovan** [comment]
lgtm

**Pull request merged**